chore: add rollup to support multiple files

Gongreg opened the pull request:
* Adding Rollup bundler to support using multiple files.
To generate output similar to currently existing one, I am using IIFE format for browser with Rune global variable and CommonJs for index.js (I think we could instead op to ES format).

Gongreg committed:
add rollup to support multiple files

bfelbo commented on dist/browser.js:
```diff
@@ -1,194 +1,213 @@
-        }
-    },
-    gameOver: function () {
+var Rune = (function () {
```
`var Rune` seems to be defined both here and below? Might be good to only define it once? :)

Gongreg replied:
Discussed in a voice chat. Since everything is inside a function, the naming of variables inside does not matter.

bfelbo commented on dist/browser.js:
```diff
@@ -1,194 +1,213 @@
-/*
```
If easy, it'd be nice to keep the same order as before to minimize the diff and keep `var Rune = { version: ... }` at the top.

Gongreg replied:
Discussed in the voice chat. The order does not matter too much, since we expect users to read the api/source code and not the dist output.

bfelbo commented on src/index.ts:
```diff
@@ -189,58 +189,10 @@ export const Rune: RuneExport = {
-    })
-  }
-})()
+setup();
```
Would be nice with a comment explaining what this is w/o having to open the file :)

bfelbo replied:
Might not be needed if we rename the file/functions.

Gongreg replied:
Renamed to `setupBrowser`.

bfelbo commented on src/index.ts:
```diff
@@ -1,6 +1,7 @@
 /*
 The SDK interface for games to interact with Rune.
 */
+import {setup} from './internal';
```
We have other internal functions (see e.g. `_randomNumberGenerator` below) that we may wanna move out into a separate file. Perhaps it'd be cleaner to name this file as `browser`, `internal/browser`, or similar? :)

Gongreg replied:
I've renamed the file/function to `setupBrowser` for now.

bfelbo reviewed (commented):
Approach looks good to me 😄

Would be great if you could test that the `browser.js` file still works. The easiest way is to copy it locally into the examples folder and make the example use the local file instead.

As we add more bundling complexity it'd be awesome to ensure that we still release ES3-compliant code. If it's easy, it would be nice to add a test for this before merging this. Perhaps using https://github.com/yowainwright/es-check or similar. If it's hard, then we can do it manually for now :)

Gongreg committed:
replace globalThis with window

Gongreg committed:
prettier

Gongreg committed:
change names

Gongreg committed:
es test

Gongreg commented:
Tested out the new bundle, the example game is working correctly with it.

Added a command to test output: `test:browserCompatibility`. Also added execution of it inside `test`. After we have a CI running for eslint/prettier etc. we can move it out from `test` so it does not run locally.

mieszko4 commented on src/setupBrowser.ts:
```diff
@@ -0,0 +1,50 @@
+  //with the iframe as user interacting. As a workaround, in the browser we will start overlay with
+  //click events disabled and display an invisible div inside the iframe above the canvas.
+  //This way the users will click on the transparent div element the very first time. We will let our client
+  //know about it with _INITIAL_OVERLAY_CLICK event and the transparent div will remove itself.
```
Nit:
```suggestion
  //know about it with BROWSER_INITIAL_OVERLAY_CLICKED event and the transparent div will remove itself.
```

Gongreg committed:
Update src/setupBrowser.ts

Co-authored-by: mieszko4 <mieszko4@gmail.com>

Gongreg committed:
build
